### PR TITLE
Fix selecting assignment in SubmitAssignment

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-13312
affects: student
release note: Fixed a bug where selecting an assignment in the share menu would get reset

Test plan:
* Delete the student app
* Install this build
* Log in as my student (s)
* Dashboard > CS 1400 > Annotate This
* Tap the link `Agenda for UCF Visit.pdf`
* Okay you have to be quick for this next part so read ahead first:
* Tap Share > Student > Tap the assignment row and before the assignments list finishes loading, quickly tap an assignment from the list
* Were you quick enough? You won't know (if it worked)
* The selected assignment should be the one you chose, not the default
* See the video in the ticket for an example of what it used to do

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by inst-danger** [comment]
#### Release Note: 
Fixed a bug where selecting an assignment in the share menu would get reset

#### Affected Apps: Student

[MBL-13312](https://instructure.atlassian.net/browse/MBL-13312)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.24% | 93.23% | 0.01%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Student/SubmitAssignmentTests/SubmitAssignmentViewControllerTests.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        XCTAssertNotNil(viewController.view)
+        env.api = URLSessionAPI()
+        env.database = database
+        env.userDefaults?.reset()
+    }
+
+    func testSelectsFirstCourseAndAssignmentByDefault() throws {
+        let presenter = try XCTUnwrap(viewController.presenter)
+        api.mock(presenter.courses, value: [.make(name: "Course 1")])
+        api.mock(GetSubmittableAssignments(courseID: "1"), value: [.make(name: "Assignment 1")])
```
Comment (shichongrui) [inline Student/SubmitAssignmentTests/SubmitAssignmentViewControllerTests.swift]: We will want to keep an eye on these to make sure this doesn't become flaky. It seemed to me that `api.mock` should have been stable but it wasn't in every case. But that could have been due to other circumstances related to specific tests.

Comment (armstrongnate) [reply Student/SubmitAssignmentTests/SubmitAssignmentViewControllerTests.swift]: Thanks, Matt. I had the same concern so I ran the tests a bunch locally. I'm going to trigger CITests again to give us one more data point.

**Pull request merged**